src/requirements/graph/processor/add-arbitrary-opt-in-choices.ts

```diff
@@ -0,0 +1,47 @@
+import GraphProcessor from './definition';
+import RequirementFulfillmentGraph from '..';
+import { CourseWithUniqueId } from '../types';
+
+export type AddArbitraryOptInChoicesParameters<Requirement extends string> = {
+  /**
+   * The mapping from course's unique ID to arbitrary opt in choices.
+   */
+  readonly userChoiceOnRequirementOverrides: {
+    readonly [uniqueId: string]: {
+      readonly arbitraryOptIn: readonly Requirement[];
+    };
+  };
+};
+
+/**
+ * Add the user's arbitrary opt in choices to the graph.
+ */
+export default class AddArbitraryOptInChoices<
+  Requirement extends string,
+  Course extends CourseWithUniqueId
+> implements GraphProcessor<Requirement, Course> {
+  private arbitraryOptInChoicesParameters: AddArbitraryOptInChoicesParameters<Requirement>;
+
+  constructor(arbitraryOptInChoicesParameters: AddArbitraryOptInChoicesParameters<Requirement>) {
+    this.arbitraryOptInChoicesParameters = arbitraryOptInChoicesParameters;
+  }
+
+  public process(graph: RequirementFulfillmentGraph<Requirement, Course>) {
+    return addArbitraryOptInChoicesParameters(graph.copy(), this.arbitraryOptInChoicesParameters);
+  }
+}
+
+export const addArbitraryOptInChoicesParameters = <
+  Requirement extends string,
+  Course extends CourseWithUniqueId
+>(
+  graph: RequirementFulfillmentGraph<Requirement, Course>,
+  { userChoiceOnRequirementOverrides }: AddArbitraryOptInChoicesParameters<Requirement>
+): RequirementFulfillmentGraph<Requirement, Course> => {
```

Comment:
OOC Is this a generic function? It's kinda hard to read/tell what's going on.

Reply:
Yes. I just realized I named it incorrectly lol. But you're right, it probably would be more clear if I labeled it as a function.

Reply:
LOL